### ISSUE TITLE
CloudFormation Module errors on existing stacks

### ISSUE THREAD

**Issue by deanwilson** [ansible/ansible#6650]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.5
##### Environment:

Running on Ubuntu managing AWS CloudFormation resources
##### Summary:

When using Ansible 1.5 with boto versions <= 2.25.0 running a playbook with multiple cloudformation resources will skip the existing stacks and create those that are missing. When upgrading boto to 2.26.0 or 2.27.0 the same playbook and version of ansible now fails if it hits an existing stack.

I think the old behaviour is the desired one.
##### Steps To Reproduce:

Working example

$ pip install boto==2.25.0
$ pip list | egrep '(ansible|boto)'

ansible (1.5)
boto (2.25.0)

$ ansible-playbook --verbose -i hosts location.playbook --verbose

PLAY [localhost] *************************************************************\* 

TASK: [Add foo] ***********************************************\* 
<localhost> REMOTE_MODULE cloudformation stack_name=foo state=present region="eu-west-1" template=foo.json

ok: [localhost] => {"changed": false, "output": "Stack is already up-to-date.", "stack_outputs": {"SNS": "arn:aws:sns:eu-west-1:"}}
- failing example

$ pip install boto==2.26.0
$ pip list | egrep '(ansible|boto)'

ansible (1.5)
boto (2.26.0)

$ ansible-playbook --verbose -i hosts location.playbook --verbose

PLAY [localhost] *************************************************************\* 

TASK: [Add foo] ***********************************************\* 
<localhost> REMOTE_MODULE cloudformation stack_name=foo state=present region="eu-west-1" template=foo.json
failed: [localhost] => {"failed": true}
msg: Stack [foo] already exists

FATAL: all hosts have already failed -- aborting

PLAY RECAP *******************************************************************\* 

localhost                  : ok=0    changed=0    unreachable=0    failed=1 
##### Expected Results:

Existing stacks should be skipped
##### Actual Results:

Existing stacks cause a Fatal.

**Comment by jsmartin** [comment]
Could you provide a couple of minimal cloudformation templates that reproduces this problem?  Maybe the original template creates one instance, and the updated template creates 2 instances?  I was not able to reproduce this issue with using these playbooks:

https://github.com/jsmartin/ansible-cf-test

On the first test I create the instance as an m1.small.  On the second round, I change the instance size to m1.medium and the update completes successfully.  This is with boto 2.27.0.

**Comment by deanwilson** [comment]
If I run the following twice in a row then I get a fatal rather than a nothing has changed with boto 2.26.0+

## ansible task

  tasks:
- name: Add SNS topic
  action: cloudformation
    stack_name={{ stack_name }}-sns-email-topic
    state=present
    region="{{region}}"
    template=sns-email-topic.json
  args:
    template_parameters:
      AutoScaleSNSTopic: "{{snsdest}}"
  register: asgsns

## cfn json

{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description" : "Creates an SNS topic",

  "Parameters" : {

```
"AutoScaleSNSTopic": {
  "Description": "Email address for notifications.",
  "Type": "String",
  "Default": "testymctest@example.org"
},

"AutoScaleSNSProtocol": {
  "Description": "The protocol that autoscale notifications should use.",
  "Type": "String",
  "Default": "email"
}
```

  },

  "Resources" : {

```
"EmailSNSTopic": {
  "Type" : "AWS::SNS::Topic",
  "Properties" : {
    "DisplayName" : "Autoscaling notifications",
    "Subscription": [
      {
       "Endpoint" : { "Ref" : "AutoScaleSNSTopic" },
       "Protocol" : { "Ref" : "AutoScaleSNSProtocol" }
      }
    ]
  }
}
```

  }
}

**Comment by jsmartin** [comment]
I can not reproduce this issue with this playbook (yours was a bit hard to read, next time, indent 4 spaces and it will make it much cleaner):

```
- hosts: localhost
  connection: local
  gather_facts: false
  vars:
    stack_name: foo
    region: us-east-1
  tasks: 
    - name: Add SNS topic 
      cloudformation:  
        stack_name={{ stack_name }}-sns-email-topic 
        state=present
        region={{ region }}
        template=sns.json 
      args: 
        template_parameters: 
          AutoScaleSNSTopic: foo@bar.com
      register: stack

    - name: show stack outputs
      debug: msg="My stack outputs are {{stack.stack_outputs}}"


{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "Creates an SNS topic",
  "Parameters": {
    "AutoScaleSNSTopic": {
      "Description": "Email address for notifications.",
      "Type": "String",
      "Default": "testymctest@example.org"
    },
    "AutoScaleSNSProtocol": {
      "Description": "The protocol that autoscale notifications should use.",
      "Type": "String",
      "Default": "email"
    }
  },
  "Resources": {
    "EmailSNSTopic": {
      "Type": "AWS::SNS::Topic",
      "Properties": {
        "DisplayName": "Autoscaling notifications",
        "Subscription": [
          {
            "Endpoint": {
              "Ref": "AutoScaleSNSTopic"
            },
            "Protocol": {
              "Ref": "AutoScaleSNSProtocol"
            }
          }
        ]
      }
    }
  }
}
```

This did produce a different error.  The stack is not able to update.  According to the AWS CloudFormation console:

"Update to resource type AWS::SNS::Topic is not supported."

**Comment by jimi-c** [comment]
This appears to have been fixed in b9ab31eb, and I can not replicate it using the current devel branch (though I do see it with v1.5.x). Since this does appear to be resolved, I'm going to go ahead and close this issue. This will be included in at least 1.6, but if you have any further questions or problems with this let us know. Thanks!
